src/http.cc

```diff
@@ -1393,6 +1393,16 @@ HttpStateData::truncateVirginBody()
     }
 }
 
```

Comment:
I mean the release-N.sgml file for the oldest of whichever vN you desire this to go to.

In the squid.conf changes section the `logformat` directive entry should have a line for the new/expanded use of `ABORTED` by `%Ss`. For example:
```
<p>The <em>ABORTED</em> value now displayed by <em>Ss</em> code to indicate
   truncated (server aborted) responses.
```

Reply:
@yadij, `%Ss` already included `ABORTED` suffixes. We even document that fact on the [wiki](https://wiki.squid-cache.org/SquidFaq/SquidLogs#squid-result-codes) -- the only place where those details are mentioned: "The response was not completed due to the connection being aborted (usually by the client)." I do not know for sure, but I suspect that some server-aborted transactions were already logged (in some Squid versions) with those `ABORTED` suffixes. 

Furthermore, these logged details are not tied to the `logformat` directive because they affect folks that do not (explicitly) use the `logformat` directive and even folks that do not log any accesses (because `%Ss` code can be sent to helpers, added to connection notes, expanded in Squid error responses, etc.)!

Finally, is it really a good idea to say something in `release-5.sgml` that is not true for any v5 code? Should not that be done when/if the change is actually ported to v5?! Imagine us documenting PR-introduced behavior in that SGML file now, and then discovering that the PR changes are not really portable to v5. What happens then? Somebody has to go in and remove those SGML changes? That does not make much sense to me. Not to mention that such vN SGML notes would effectively require vN maintainer to make a backporting decision at master PR review time, increasing the number of potential conflict points.

Given all of the above, are you absolutely sure that we should modify `release-5.sgml` to give these changes a chance for a v5 backport? If you are, I will add those changes (either in this PR if you respond promptly or in a dedicated PR otherwise).

Reply:
> @yadij, `%Ss` already included `ABORTED` suffixes. We even document that fact on the [wiki](https://wiki.squid-cache.org/SquidFaq/SquidLogs#squid-result-codes) -- the only place where those details are mentioned: "The response was not completed due to the connection being aborted (usually by the client)." I do not know for sure, but I suspect that some server-aborted transactions were already logged (in some Squid versions) with those `ABORTED` suffixes.

AFAIK the situation was rare edge cases, now it is about to become certain / common. That will be a bit of a surprise for many admin.

> 
> Furthermore, these logged details are not tied to the `logformat` directive ...

The %codes are documented with logformat directive and the other directives sharing have references to that directives documentation. I do not see an issue here.


> Finally, is it really a good idea to say something in `release-5.sgml` that is not true for any v5 code?

Patches should always include documentation updates associated with their change.

I have been very lax in the past, doing it myself as followup. With improved maintenance automation coming this practice of including all docs is one that needs to be more rigorously followed.

Please also remember I simply mentioned that v5 was a candidate for backport (as in I expect it aceiptible). That choice remains with the author. If they want it in v5, they should also be willing to do any adjustments for backport. If not just document in later vN release notes.


> Should not that be done when/if the change is actually ported to v5?! Imagine us documenting PR-introduced behavior
> in that SGML file now, and then discovering that the PR changes are not really portable to v5. What happens then?
> Somebody has to go in and remove those SGML changes?

Yes. Been there, done that. No problems.

> That does not make much sense to me. Not to mention that such vN SGML notes would effectively require vN maintainer to make a backporting decision at master PR review time, increasing the number of potential conflict points.

That is already a reviewer consideration:
```
 * Is PR a bug or security fix?
  - if yes, which versions are affected?
    + if stable, ensure / request style and documentation are compatible for backport
```

> 
> Given all of the above, are you absolutely sure that we should modify `release-5.sgml` to give these changes a chance for a v5 backport? If you are, I will make those modifications (either in this PR if you respond promptly or in a dedicated PR otherwise).

I am sure that I am okay with PR change being backported. I left the choice to the author whether they included the preparation for that, or just prepared it for v6 (or even v7).

Reply:
> Patches should always include documentation updates associated with their change. ... this practice of including all docs is one that needs to be more rigorously followed..

I agree that a PR making a change should, by default, update documentation. However, I do not think a PR should update _unrelated_ documentation (e.g., we should not update vX release notes in a PR against master/vY, where X is not Y).

The need to manually update vN release notes during a backport to vN is certainly a pain. However, a correct solution for that problem differs from asking PR authors to update whatever vN release notes they think should be updated for _future_ backports (because the timing and even existence of those future backports is uncertain at the master PR time and because the PR is not updating those vN branches).

I have not checked all the details, but I suspect that a backport should copy the relevant entries from the master release notes and master change log. Adopting my recent [suggestion](https://github.com/squid-cache/squid/pull/1245#pullrequestreview-1269568901) to add commit SHAs to such entries would help automate most of such copying. More work is needed to make them fully automated -- I can make specific recommendations if anybody is interested in implementing them.

> That is already a reviewer consideration (for me at least): ... ensure / request style and documentation are compatible for backport

Please do _not_ ask for vN-specific documentation changes in PRs modifying code in the _master_ branch. Documentation stored in the master branch is not the right place to document possible future changes in existing release branches. Needless to say, a vN maintainer has the right to make the corresponding vN release notes update a _precondition_ for accepting the changes into vN. That documentation update should be done as a PR against that vN branch. Ideally, that vN PR should contain all relevant code and documentation changes bundled together -- representing a single/atomic change that preserves code-documentation synchronization.

----

Even though I disagree that we should document this particular change in release notes about the `logformat` directive, I added such documentation in PR branch commit f54d4de. Hopefully, that addition will make any future backports easier.

I am clearing this PR for merge.

> Amos at https://github.com/squid-cache/squid/pull/1244#issuecomment-1399385529: Marked as feature due to missing release notes entry for logformat codes changed behaviour/output.

I have not removed the `feature` label from this PR despite release notes update because I do not fully understand what that label means. Please remove that label if it should be removed. I continue to plea for a better label name and documentation, so that each of us can maintain that label (or at least understand, without questions/discussions, the implications of you adding/removing that label).